cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java

```diff
@@ -623,13 +648,16 @@ public void setCoordinator(Coordinator coordinator) {
           result = SyncClientAdaptor
               .getUnregisteredMeasurements(client, partitionGroup.getHeader(), seriesList);
         } else {
-          SyncDataClient syncDataClient =
-              metaGroupMember.getClientProvider().getSyncDataClient(node,
-                  RaftServer.getReadOperationTimeoutMS());
-          result = syncDataClient.getUnregisteredTimeseries(partitionGroup.getHeader(), seriesList);
-          ClientUtils.putBackSyncClient(syncDataClient);
+          SyncDataClient syncDataClient = null;
+          try {
+            syncDataClient = metaGroupMember.getClientProvider().getSyncDataClient(node,
+                RaftServer.getReadOperationTimeoutMS());
+            result = syncDataClient
+                .getUnregisteredTimeseries(partitionGroup.getHeader(), seriesList);
+          } finally {
+            ClientUtils.putBackSyncClient(syncDataClient);
```

Comment:
What about getSyncDataClient throws exception and syncDataClient is null and at this circumstance is still null?

Reply:
null client value can not be put back to the client pool, this is guaranteed in function putBackSyncClient in ClientUtils,
if the client is null, the two if in the judgment condition is not tenable, so it will not be put back.